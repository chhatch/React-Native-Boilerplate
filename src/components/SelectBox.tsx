import React, { useRef } from 'react'
import { ViewStyle } from 'react-native'
import { Picker } from '@react-native-picker/picker'

type Option = { label: string; value: string }
type Options = Option[]

interface ISelectBox {
    defaultValue: string
    onChange: Function
    options: Options
    prompt: string
    style: ViewStyle
}
const SelectBox = ({
    defaultValue,
    onChange,
    options,
    prompt,
    style,
}: ISelectBox) => {
    const val = useRef(defaultValue)
    return (
        <Picker
            selectedValue={val.current}
            style={style}
            onValueChange={(itemValue, itemIndex) => {
                val.current = itemValue
                onChange(itemValue)
            }}
            prompt={prompt}
        >
            {options.map(({ label, value }: IOption, index: number) => (
                <Picker.Item label={label} value={value} key={index} />
            ))}
        </Picker>
    )
}

export default SelectBox
