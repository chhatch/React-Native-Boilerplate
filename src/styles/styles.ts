import { Dimensions, StyleSheet, TextStyle, ViewStyle } from 'react-native'

const width = Dimensions.get('window').width
const height = Dimensions.get('window').height

export const styles = StyleSheet.create({
    input: {
        height: 50,
        backgroundColor: '#ededed',
        width: '33%',
    },
    container: {
        flex: 1,
        justifyContent: 'center',
        alignItems: 'center',
        backgroundColor: '#F5FCFF',
    },
    button: {
        marginBottom: 20,
    },
    title: {
        fontSize: 20,
        fontWeight: 'bold',
    },
    separator: {
        marginVertical: 30,
        height: 1,
        width: '80%',
    },
    head: { height: 40, backgroundColor: '#f1f8ff' },
    text: { margin: 6 },
    loading: {
        position: 'absolute',
        left: 0,
        right: 0,
        top: 0,
        bottom: 0,
        alignItems: 'center',
        justifyContent: 'center',
    },
    formError: {
        marginBottom: 20,
    },
})

interface Alerts {
    alertContainer: ViewStyle
    error: TextStyle
    success: TextStyle
}

export const alertStyles = StyleSheet.create<Alerts>({
    alertContainer: {
        marginTop: 63,
        position: 'absolute',
        width: width,
        zIndex: 999,
    },
    error: {
        backgroundColor: 'red',
        fontSize: 20,
    },
    success: {
        backgroundColor: 'green',
        fontSize: 20,
    },
})
