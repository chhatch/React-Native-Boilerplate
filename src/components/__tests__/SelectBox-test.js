import * as React from 'react'
import renderer from 'react-test-renderer'

import SelectBox from '../SelectBox'
import { styles } from '../../styles/styles'

const message = 'Snapshot test!'
const style = styles.formError


it(`renders a select box`, () => {
    const tree = renderer
        .create(
            <SelectBox
                defaultValue={'USER'}
                onChange={() => console.log('this is a test')}
                options={[
                    { label: 'GLOBAL_ADMIN', value: 'GLOBAL_ADMIN' },
                    { label: 'ADMIN', value: 'ADMIN' },
                    { label: 'USER', value: 'USER' },
                ]}
                prompt="organization"
                style={styles.input}
            />
        )
        .toJSON()
    expect(tree).toMatchSnapshot()
})
