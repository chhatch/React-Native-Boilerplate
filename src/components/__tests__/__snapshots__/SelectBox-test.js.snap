// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a select box 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ededed",
      "height": 50,
      "width": "33%",
    }
  }
>
  <RNCPicker
    items={
      Array [
        Object {
          "label": "GLOBAL_ADMIN",
          "testID": undefined,
          "textColor": undefined,
          "value": "GLOBAL_ADMIN",
        },
        Object {
          "label": "ADMIN",
          "testID": undefined,
          "textColor": undefined,
          "value": "ADMIN",
        },
        Object {
          "label": "USER",
          "testID": undefined,
          "textColor": undefined,
          "value": "USER",
        },
      ]
    }
    onChange={[Function]}
    selectedIndex={2}
    style={
      Array [
        Object {
          "height": 216,
        },
        undefined,
      ]
    }
  />
</View>
`;
